editor/editor_properties.cpp

```diff
@@ -2183,7 +2183,55 @@ void EditorPropertyResource::_menu_option(int p_which) {
 		case OBJ_MENU_NEW_SCRIPT: {
 
 			if (Object::cast_to<Node>(get_edited_object())) {
-				EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				if (get_edited_property() == "script") {
+					EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				} else {
+					// Copied from editor/scene_tree_dock.cpp#L380 and edited for our purposes
+					// There's a good chance we could merge this back into that code with some modifications
```

Comment:
As mentioned here, there's a good chance we could merge this back into SceneTreeDock's _script_created code, though it'll require some modifications.

Reply:
It depends on how much modification is needed. If the final result is less or ~same amount of code, that's fine, but if it implies adding a lot of abstraction just to reduce 15 lines of code duplication, it's likely not worth it.

What are your ideas to merge this?

Reply:
My idea to merge this is to edit _script_created and _tool_selected (both from SceneTreeDock) to take a "property" parameter, which will be provided as a signal bind. When it is set, the script will be attached to that property after creation (for _script_created) or used to generate the "inherits" and "path" bits (for _tool_selected).

Ideally, this method will be used whether we're attaching the script to the `script` property, or to another Script-type property. (taking a look at object.cpp, it seems it updates the script instance when the `script` property is updated via `set()`, but I will test that).

Reply:
I will work on merging this into scene_tree_dock.cpp after I finish working on all the other problems here.